test/e2e-tests.sh

```diff
@@ -25,13 +25,148 @@
 
 source $(dirname $0)/../scripts/e2e-tests.sh
 
+# Read metadata.json and get value for key
+# Parameters: $1 - Key for metadata
+function get_meta_value() {
+  # local metadata="${ARTIFACTS}/metadata.json"
+  # cat ${metadata} | grep -oEi "\"${1}\" ?: ?\".*\"" | cut -d "\"" -f 4
+  go run knative.dev/test-infra/test/metahelper --get $@
+}
+
 function knative_setup() {
   start_latest_knative_serving
 }
 
+# Run a prow-cluster-operation tool, installing it first if necessary.
+#             $1..$n - parameters passed to the tool.
+function run_prow_cluster_tool() {
+  go run knative.dev/test-infra/test/prow-cluster-operation $@
+}
+
+# Add function call to trap
+# Parameters: $1 - Function to call
+#             $2...%n - Signals for trap
+function add_trap {
+  local cmd=$1
+  shift 1
+  for trap_signal in "$@"; do
+    local current_trap="$(trap -p $trap_signal | cut -d\' -f2)"
+    local new_cmd=$cmd
+    [[ -n "${current_trap}" ]] && new_cmd="${current_trap};${new_cmd}"
+    trap -- "${new_cmd}" $trap_signal
+  done
+}
+
+# Override create_test_cluster in scripts/e2e-tests.sh
+# Create test cluster with cluster creation lib and write metadata in ${ARTIFACT}/metadata.json
+function create_test_cluster() {
+  # Fail fast during setup.
+  set -o errexit
+  set -o pipefail
+
+  header "Creating test cluster"
+
+  local creation_args=""
+  (( SKIP_ISTIO_ADDON )) || creation_args+=" --addons istio"
+  [[ -n "${GCP_PROJECT}" ]] && creation_args+=" --project ${GCP_PROJECT}"
+  echo "creation args: ${creation_args}"
+  run_prow_cluster_tool --create ${creation_args} || fail_test "failed creating test cluster"
+  # Should have kubeconfig set already
+  local k8s_cluster=$(kubectl config current-context)
+  [[ -z "${k8s_cluster}" ]] && abort "kubectl must have been set at this point"
+  # Add protection before trapping removal
+  is_protected_cluster ${k8s_cluster} && \
+    abort "kubeconfig context set to ${k8s_cluster}, which is forbidden"
+
+  # Since this function is called assuming cluster creation, force removing
```

Comment:
Not sure I understand this comment. This function is "create_test_cluster", why one would assume cluster creation? What's the importance of "cluster-remover forces cleaning up everything", isn't it expected? Finally, --delete doesn't wait for the operation to complete, why the TODO?